plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -918,6 +934,27 @@ public static Map<String, String> createTableProperties(ConnectorTableMetadata t
                 .buildOrThrow();
     }
 
+    private static Optional<Entry<String, String>> resolveWriteCompressionCodecProperty(ConnectorSession session, IcebergFileFormat fileFormat)
+    {
+        return switch (fileFormat) {
+            case PARQUET -> getCompressionCodec(session).getParquetCompressionCodec()
```

Comment:
`getCompressionCodec(session).getParquetCompressionCodec()` returns an `Optional` 

Do we have coverage in the tests for such a use case?

Reply:
> Do we have coverage in the tests for such a situation?

[TestIcebergSparkCompatibility.testSparkReadingTrinoCompressedData](https://github.com/trinodb/trino/blob/1f808b1667712b7d56ae1e4f04f7bf5170c67d81/testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java#L1632) - runs with all file _format X codec_ combinations, but I don't think we test for `Optional.empty()` case - looks like all the code in `trino-iceberg` calling `getAvroCompressionKind()` or `getParquetCompressionCodec()` treats empty optional as an error, I will do the same here. Thank you for pointing this out!
![image](https://github.com/user-attachments/assets/c87af993-12a8-40f3-8d63-403cd8580019)

Reply:
Pushed changes, we actually assert on that in `BaseIcebergConnectorTest.testCreateTableWithCompressionCodec` - Optional is empty and error is expected. Prior to my changes, the error was triggered upstream.